src/term.c

```diff
@@ -797,6 +827,32 @@ static int cmd_send(PROGRAMMER *pgm, AVRPART *p, int argc, char *argv[]) {
 
 
 static int cmd_erase(PROGRAMMER *pgm, AVRPART *p, int argc, char *argv[]) {
+  if (argc > 4 || argc == 3) {
+    msg_error("Usage: erase <memory> <addr> <len>\n");
+    return -1;
+  }
+
+  if (argc > 1) {
+    char *memtype = argv[1];
+    AVRMEM *mem = avr_locate_mem(p, memtype);
+    if (mem == NULL) {
+      pmsg_error("(erase) %s memory type not defined for part %s\n", argv[1], p->desc);
+      return -1;
+    }
+    char *args[6] = {"write", memtype, "", "", "0xff", "..."};
```

Comment:
This is not wrong, but tradition has it that argv lists are terminated by NULL, so just for superstition, I'd write `char *args[] = {"write", memtype, "", "", "0xff", "...", NULL};` Admittedly, it's unlikely that the write code ever looks for the terminating NULL.

Reply:
Thanks for your input, I wasn't aware of this so if I add a NULL to the args array, should I pass 6 or 7 as the argc number to cmd_write?

Reply:
argc must be 6 (no change)

Reply:
https://en.cppreference.com/w/cpp/language/main_function

argc	-	Non-negative value representing the number of arguments passed to the program from the environment in which the program is run.

argv	-	Pointer to the first element of an array of argc + 1 pointers, of which the last one is null and the previous ones, if any, point to null-terminated multibyte strings that represent the arguments passed to the program from the execution environment. If argv[0] is not a null pointer (or, equivalently, if argc > 0), it points to a string that represents the name used to invoke the program, or to an empty string.

Reply:
I had a look at the `tokenize()` function that provides the argc, argv pair for all the commands in the terminal. Indeed, that uses a terminating NULL (as convention has it). So better to change that line to `char *args[] = {"write", memtype, "", "", "0xff", "...", NULL};`